### DISCUSSION TITLE
Modify all data before going in cache

### DISCUSSION THREAD

**Discussion by cristobalbahe** [urql-graphql/urql#1981] [Q&A] [answered]
Hello everyone,

I am having a tough time understanding exchanges and I don't know how to get where I am trying to get.

I am using graphql-crunch on my server, which is a library to optimize JSON responses by minimizing duplication and improving compressibility.  The thing is that what the server sends to my app is an array resulting from crunching the data instead of the regular nested JSON response so if it goes directly into the graphcache nothing will work.

If I could uncrunch the data on the client before passing it to the graphcache exchange, I could benefit both from graphql-crunch and caching.

Do anyone knows how I could do this?

Thank you so much in advance.

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

You'll need to put an exchange between the `fetch` and `graphCache` exchange like `exchanges: [dedup, cache, yourExchange, fetch]`.

An exchange could look like the following:

```
import { pipe, map } from 'wonka';

export function crunchExchange({ forward }) {
  return function($ops) {
    pipe(
      $ops,
      forward,
      map(handleCrunchResult)
    )
  }
}
```

**Reply by cristobalbahe** [reply]
Saved my day! 

It works perfectly, thank you so much!
